common/util.go

```diff
@@ -149,3 +150,18 @@ func GetConnectionAddr(ctx context.Context) string {
 	}
 	return from
 }
+
+// RatToFixed converts a big.Rat into a fixed point number represented as int64
+// according to supplied scale factor and maximum decimalplaces
+func RatToFixed(rat *big.Rat, scalingFactor int64, maxDecimalPlaces int) (int64, error) {
```

Comment:
Do we anticipate variation in how this is used, or could we hard-code the last two params? The advantage I see is there's less room for error when converting pricing info (eg, someone down the road uses a scaling factor of 10000 rather than 1000).

Reply:
I don't know going forward, but I constructed it this way out of possible flexibility in the future. 

I did some playing with the scaling factor (I was a proponent of using fixed point 52.12) and with the current setup we run into "collisions" starting at a 1/36 big.Rat. Whether that's sufficient will be dependant on real market prices set by users.

Reply:
The values of scalingFactor/maxDecimalPlaces are fine to me, the idea is more towards making the function itself misuse-resistant. (Also, as @yondonfu mentioned, changing this in a backwards-compatible manner would be difficult.) Renaming the function may also help clarify its intended usage. In particular, this is what I was thinking:

```
func PriceToFixed(price *big.Rat) {
  scalingFactor := 1000
  maxDecimalPlaces := 3
  ... rest of the code ...
}
```

Later, if we have a need for further specialization on those params, we can split into helpers as needed - exposing those right now seems premature, and in particular the `maxDecimalPlaces` param is really just the implementation leaking through the API.

Reply:
In my opinion utility functions should be named according to the functionality they provide, regardless of context , in this case converting a big.Rat to a fixed point. 

RatToFixed can be used to convert any big.Rat to a fixed point representation, not limited to only pricing information represented as a big.Rat. 

If we think we'll only be converting those price specific big.Rats to fixed points ever, then I would agree hardcoding the scalingfactor/maxDecimals and renaming it to `PriceToFixed`
That's more the thought process that was on my mind when writing this the way I did.

Reply:
> Later, if we have a need for further specialization on those params, we can split into helpers as needed - exposing those right now seems premature

Agreed. Even if we do end up needing to use different values for `scalingFactor` (i.e. if we need different fixed point number types with difference precision in other areas of the codebase) in the future, we would probably want to abstract `scalingFactor` away from the caller to avoid programmer error (i.e. passing in an incorrect value for `scalingFactor`).

This is premature, but in the future it could look something like:

```
func PriceToFixed(price *big.Rat) {
    scalingFactor := 1000
    return ratToFixed(price, scalingFactor)
}
```

In this future scenario, where there might be a use case for different values of `scalingFactor`, we could refactor and move the current logic of `RatToFixed()` down one level on the abstraction ladder into the `ratToFixed()` helper which *does* expose `scalingFactor` as a parameter. However, `ratToFixed()` would only ever be used internally within the package, and package users would only use the exported functions that abstract `scalingFactor` away.

For now, IMO just having `PriceToFixed(rat *big.Rat)` while omitting `scalingFactor` from the parameters is fine.